cmd/kube-scheduler/app/server.go

```diff
@@ -207,7 +207,11 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 	if cc.DynInformerFactory != nil {
 		cc.DynInformerFactory.WaitForCacheSync(ctx.Done())
 	}
-
+	// Wait for all handlers to sync (all items in the initial list delivered) before scheduling.
+	if err := sched.WaitForHandlersSync(ctx); err != nil {
+		logger.Error(err, "error occurred during WaitForHandlersSync")
```

Comment:
do we keep going despite it fails?

Reply:
> do we keep going despite it fails?

Nope, we return the error if it fails here. Plz see the new commit.